main.py

```diff
@@ -2303,40 +2309,69 @@ def on_modified_async(self):
-                        if paramDocs:
-                            formatted.append("**{}**\n".format(parameter.get('label')))
-                            formatted.append("* *{}*\n".format(paramDocs))
-
-                formatted.append(signature.get('documentation'))
-
-                mdpopups.show_popup(
-                    self.view,
-                    preserve_whitespace("\n".join(formatted)),
-                    css=".mdpopups .lsp_signature { margin: 4px; } .mdpopups p { margin: 0.1rem; }",
-                    md=True,
-                    flags=sublime.HIDE_ON_MOUSE_MOVE_AWAY,
-                    location=point,
-                    wrapper_class="lsp_signature",
-                    max_width=800)
+            self._config = config_for_scope(self.view)
+            self.signatures = response.get("signatures")
+            self.active_signature = response.get("activeSignature", -1)
+            debug("got signatures, active is", len(self.signatures), self.active_signature)
+            if len(self.signatures) > 0 and self._config:
+                mdpopups.show_popup(self.view,
+                                    self._build_popup_content(),
+                                    css=self.__class__.css,
+                                    md=True,
+                                    flags=sublime.HIDE_ON_MOUSE_MOVE_AWAY,
+                                    location=point,
+                                    wrapper_class=self.__class__.wrapper_class,
+                                    max_width=800,
+                                    on_hide=self._on_hide)
+                self._visible = True
+
+    def on_query_context(self, key, _, operand, __):
+        if key != "lsp.signature_help":
+            return False  # Let someone else handle this keybinding.
+        elif not self._visible:
+            return False  # Let someone else handle this keybinding.
+        elif self.active_signature not in range(0, len(self.signatures)):
+            return False  # Let someone else handle this keybinding.
+        else:
+            # We use the "operand" for the number -1 or +1. See the keybindings.
+            self.active_signature += operand
+            self.active_signature %= len(self.signatures)
+            mdpopups.update_popup(self.view,
+                                  self._build_popup_content(),
+                                  css=self.__class__.css,
+                                  md=True,
+                                  wrapper_class=self.__class__.wrapper_class)
+            return True  # We handled this keybinding.
+
+    def _on_hide(self):
+        self._visible = False
+
+    def _build_popup_content(self) -> str:
+        signature = self.signatures[self.active_signature]
+        debug("active signature", signature)
+        formatted = []
+        if len(self.signatures) > 1:
+            label = "```{}\n{}/{} {}\n```".format(self._config.languageId, str(self.active_signature + 1),
```

Comment:
It might be less confusing to the user if the `1 / 2` was not formatted as a code block like the signature itself. Would it be okay to show it below or above the code block?